cursor.c

```diff
@@ -33,7 +89,7 @@ uint16_t st_cursor_get_keycode(st_cursor_t *cursor)
     if (!keyaction) {
         return KC_NO;
     }
-    if (cursor->cursor_pos.as_output_buffer &&
+    if (cursor->cursor_pos.as_output &&
```

Comment:
Are you sure about the if/else logic here?
Currently I find it hard if this is a code optimization or a logic error.
Would be more clear if we handle the input/voutput cases separately:
```
if (!cursor->cursor_pos.as_output) {
    return keyaction->keypressed;
}
// handle vout case
```
Thoughts?

Reply:
This is correct, but it might be a bit clearer if I reverse the if and use early return.

If the cursor is not an output_cursor OR it's an output cursor on a regular (non-rule matching keypress), then we simply use the keypress from the key_buffer. If it is an output cursor AND the currently focused key in the buffer is an action, then we need to grab the char from the completion using the sub_index.